x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -370,42 +370,17 @@ public void testAnotherCountAllWithFilter() {
         assertThat(expected.toString(), is(esStatsQuery.query().toString()));
     }
 
-    /**
-     * Expected
-     * ProjectExec[[c{r}#3, c{r}#3 AS call, c_literal{r}#7]]
-     * \_LimitExec[1000[INTEGER]]
-     *   \_AggregateExec[[],[COUNT([2a][KEYWORD]) AS c, COUNT(1[INTEGER]) AS c_literal],FINAL,null]
-     *     \_ExchangeExec[[count{r}#18, seen{r}#19, count{r}#20, seen{r}#21],true]
-     *       \_EsStatsQueryExec[test], stats[Stat[name=*, type=COUNT, query=null], Stat[name=*, type=COUNT, query=null]]],
-     *         query[{"esql_single_value":{"field":"emp_no","next":{"range":{"emp_no":{"gt":10010,"boost":1.0}}},
-     *         "source":"emp_no > 10010@2:9"}}][count{r}#23, seen{r}#24, count{r}#25, seen{r}#26], limit[],
```

Comment:
`| stats c = count(), call = count(*), c_literal = count(1)`  produced an `EsStatsQueryExec` that wasn't plannable by the LocalExecutionPlanner (`EsStatsQuery currently supports only one field statistic`).

I think a way to properly fix this is:
- normalize count(*), count(1), count(), count("foobar") all to count(*) as part of surrogate substitution
- deduplicate the counts - this requires refactoring `SubstituteExpressions` as its deduplication doesn't currently work well enough for this approach, and there's overlap with the deduplication happening in `ReplaceStatsAggExpressionWithEval`.

Reply:
Once the count normalization happens, the deduplication should kick in (since all counts will be the same) and this type of query should work.
The critical bit here is normalization so count(*), count(1), count(), count(*) and count(1+1+2) all fold back to the same expression of count(1).